<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindowClass</class>
 <widget class="QMainWindow" name="MainWindowClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1024</width>
    <height>786</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>HIPPIE</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="RenderWidget" name="glwidget" native="true">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="focusPolicy">
       <enum>Qt::ClickFocus</enum>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1024</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionHelp"/>
    <addaction name="separator"/>
    <addaction name="actionAbout"/>
   </widget>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionOpen_Scene"/>
    <addaction name="actionClose_Scene"/>
    <addaction name="separator"/>
    <addaction name="actionShow_Stats"/>
    <addaction name="separator"/>
    <addaction name="actionScreenshot"/>
    <addaction name="separator"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuBVH">
    <property name="title">
     <string>BVH</string>
    </property>
    <addaction name="actionLBVH"/>
    <addaction name="actionHLBVH"/>
    <addaction name="actionSBVH"/>
    <addaction name="actionPLOC"/>
    <addaction name="actionTR"/>
    <addaction name="actionATR"/>
    <addaction name="actionInsertion"/>
    <addaction name="separator"/>
    <addaction name="actionRefit"/>
    <addaction name="actionRebuild"/>
    <addaction name="separator"/>
   </widget>
   <widget class="QMenu" name="menuRendering">
    <property name="title">
     <string>Rendering</string>
    </property>
    <addaction name="actionPrimary_Rays"/>
    <addaction name="actionShadow_Rays"/>
    <addaction name="actionAO_Rays"/>
    <addaction name="actionPath_Rays"/>
    <addaction name="actionPseudocolor_Rays"/>
    <addaction name="actionThermal_Rays"/>
    <addaction name="separator"/>
    <addaction name="actionParameters"/>
    <addaction name="actionLight"/>
    <addaction name="separator"/>
    <addaction name="actionReset_Camera"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuBVH"/>
   <addaction name="menuRendering"/>
   <addaction name="menuHelp"/>
  </widget>
  <action name="actionAbout">
   <property name="checkable">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>About GPU Ray Tracer</string>
   </property>
  </action>
  <action name="actionOpen_Scene">
   <property name="text">
    <string>Open Scene</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
  </action>
  <action name="actionHelp">
   <property name="text">
    <string>Help</string>
   </property>
  </action>
  <action name="actionClose_Scene">
   <property name="text">
    <string>Close Scene</string>
   </property>
  </action>
  <action name="actionHLBVH">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>HLBVH</string>
   </property>
  </action>
  <action name="actionSettings">
   <property name="text">
    <string>Settings</string>
   </property>
  </action>
  <action name="actionShow_Stats">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Show Stats</string>
   </property>
  </action>
  <action name="actionScreenshot">
   <property name="text">
    <string>Screenshot</string>
   </property>
  </action>
  <action name="actionPrimary_Rays">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Primary Rays</string>
   </property>
  </action>
  <action name="actionAO_Rays">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>AO Rays</string>
   </property>
  </action>
  <action name="actionDiffuse_Rays">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Diffuse Rays</string>
   </property>
  </action>
  <action name="actionReset_Camera">
   <property name="text">
    <string>Default Camera</string>
   </property>
  </action>
  <action name="actionShadow_Rays">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Shadow Rays</string>
   </property>
  </action>
  <action name="actionRefit">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Refit</string>
   </property>
  </action>
  <action name="actionLoad_BVH">
   <property name="text">
    <string>Load BVH</string>
   </property>
  </action>
  <action name="actionLoaded_BVH">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Loaded BVH</string>
   </property>
  </action>
  <action name="actionImport">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Import</string>
   </property>
  </action>
  <action name="actionPseudocolor_Rays">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Pseudocolor Rays</string>
   </property>
  </action>
  <action name="actionParameters">
   <property name="text">
    <string>Parameters</string>
   </property>
  </action>
  <action name="actionLight">
   <property name="text">
    <string>Light</string>
   </property>
  </action>
  <action name="actionThermal_Rays">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Thermal Rays</string>
   </property>
  </action>
  <action name="actionPath_Rays">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Path Rays</string>
   </property>
  </action>
  <action name="actionPLOC">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>PLOC</string>
   </property>
  </action>
  <action name="actionLBVH">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>LBVH</string>
   </property>
  </action>
  <action name="actionATR">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>ATR</string>
   </property>
  </action>
  <action name="actionTR">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>TR</string>
   </property>
  </action>
  <action name="actionRebuild">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Rebuild</string>
   </property>
  </action>
  <action name="actionOptimize">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Optimize</string>
   </property>
  </action>
  <action name="actionInsertion">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Insertion</string>
   </property>
  </action>
  <action name="actionSBVH">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>SBVH</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>RenderWidget</class>
   <extends>QWidget</extends>
   <header>RenderWidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>actionAbout</sender>
   <signal>triggered()</signal>
   <receiver>MainWindowClass</receiver>
   <slot>showAboutDialog()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>299</x>
     <y>199</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionExit</sender>
   <signal>triggered()</signal>
   <receiver>MainWindowClass</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>299</x>
     <y>199</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionOpen_Scene</sender>
   <signal>triggered()</signal>
   <receiver>MainWindowClass</receiver>
   <slot>openScene()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>511</x>
     <y>394</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionClose_Scene</sender>
   <signal>triggered()</signal>
   <receiver>MainWindowClass</receiver>
   <slot>closeScene()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>511</x>
     <y>394</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionShow_Stats</sender>
   <signal>triggered(bool)</signal>
   <receiver>MainWindowClass</receiver>
   <slot>enableStats(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>511</x>
     <y>394</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionScreenshot</sender>
   <signal>triggered()</signal>
   <receiver>MainWindowClass</receiver>
   <slot>takeScreenshot()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>511</x>
     <y>394</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionPrimary_Rays</sender>
   <signal>triggered()</signal>
   <receiver>MainWindowClass</receiver>
   <slot>setPrimaryRays()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>511</x>
     <y>394</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionAO_Rays</sender>
   <signal>triggered()</signal>
   <receiver>MainWindowClass</receiver>
   <slot>setAORays()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>511</x>
     <y>394</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionHelp</sender>
   <signal>triggered()</signal>
   <receiver>MainWindowClass</receiver>
   <slot>showHelpDialog()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>511</x>
     <y>394</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionReset_Camera</sender>
   <signal>triggered()</signal>
   <receiver>MainWindowClass</receiver>
   <slot>resetCamera()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>511</x>
     <y>394</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionPrimary_Rays</sender>
   <signal>triggered()</signal>
   <receiver>MainWindowClass</receiver>
   <slot>setPrimaryRays()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>511</x>
     <y>394</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionShadow_Rays</sender>
   <signal>triggered()</signal>
   <receiver>MainWindowClass</receiver>
   <slot>setShadowRays()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>511</x>
     <y>392</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionPseudocolor_Rays</sender>
   <signal>triggered()</signal>
   <receiver>MainWindowClass</receiver>
   <slot>setPseudocolorRays()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>511</x>
     <y>392</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionParameters</sender>
   <signal>triggered()</signal>
   <receiver>MainWindowClass</receiver>
   <slot>showRenderParamsWindow()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>511</x>
     <y>392</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionLight</sender>
   <signal>triggered()</signal>
   <receiver>MainWindowClass</receiver>
   <slot>showLightWindow()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>511</x>
     <y>392</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionHLBVH</sender>
   <signal>triggered()</signal>
   <receiver>MainWindowClass</receiver>
   <slot>setHLBVH()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>511</x>
     <y>392</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionThermal_Rays</sender>
   <signal>triggered()</signal>
   <receiver>MainWindowClass</receiver>
   <slot>setThermalRays()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>511</x>
     <y>392</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionPath_Rays</sender>
   <signal>triggered()</signal>
   <receiver>MainWindowClass</receiver>
   <slot>setPathRays()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>511</x>
     <y>392</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionPLOC</sender>
   <signal>triggered()</signal>
   <receiver>MainWindowClass</receiver>
   <slot>setPLOC()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>511</x>
     <y>392</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionLBVH</sender>
   <signal>triggered()</signal>
   <receiver>MainWindowClass</receiver>
   <slot>setLBVH()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>511</x>
     <y>392</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionATR</sender>
   <signal>triggered()</signal>
   <receiver>MainWindowClass</receiver>
   <slot>setATR()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>511</x>
     <y>392</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionTR</sender>
   <signal>triggered()</signal>
   <receiver>MainWindowClass</receiver>
   <slot>setTR()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>511</x>
     <y>392</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionRefit</sender>
   <signal>triggered()</signal>
   <receiver>MainWindowClass</receiver>
   <slot>setRefit()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>511</x>
     <y>392</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionRebuild</sender>
   <signal>triggered()</signal>
   <receiver>MainWindowClass</receiver>
   <slot>setRebuild()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>511</x>
     <y>392</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionInsertion</sender>
   <signal>triggered()</signal>
   <receiver>MainWindowClass</receiver>
   <slot>setInsertion()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>511</x>
     <y>392</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSBVH</sender>
   <signal>triggered()</signal>
   <receiver>MainWindowClass</receiver>
   <slot>setSBVH()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>511</x>
     <y>392</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>showAboutDialog()</slot>
  <slot>setPrimaryRays()</slot>
  <slot>setAORays()</slot>
  <slot>setDiffuseRays()</slot>
  <slot>openScene()</slot>
  <slot>closeScene()</slot>
  <slot>enableStats(bool)</slot>
  <slot>takeScreenshot()</slot>
  <slot>showHelpDialog()</slot>
  <slot>resetCamera()</slot>
  <slot>setShadowRays()</slot>
  <slot>setPseudocolorRays()</slot>
  <slot>showRenderParamsWindow()</slot>
  <slot>showLightWindow()</slot>
  <slot>setHLBVH()</slot>
  <slot>setThermalRays()</slot>
  <slot>setPathRays()</slot>
  <slot>setPLOC()</slot>
  <slot>setLBVH()</slot>
  <slot>setATR()</slot>
  <slot>setTR()</slot>
  <slot>setRefit()</slot>
  <slot>setRebuild()</slot>
  <slot>setInsertion()</slot>
  <slot>setSBVH()</slot>
 </slots>
</ui>
